### PR TITLE
burrow rework

### PR DESCRIPTION
**Pull request by ITCups**
- new sprites
- burrows now separated into closed ones (cracks) and open, closed burrows are often ignored when choosing target for migrations
- burrows can be closed with explosions
- burrow can be closed with steel sheets
- spawns scrap pile with random loot if closed with digging tool

removed unnecessary loging

### CODE REVIEW

File: code/game/objects/effects/effect_system.dm [hunk 1]
```diff
@@ -27,6 +28,10 @@ would spawn and follow the beaker, even if it is carried or thrown.
 			M.Turn(rand(0,360))
 
 		transform = M
+	if(random_offset)
```
Comment (LethalGhost) [inline code/game/objects/effects/effect_system.dm]: Shouldn't that be in `Initialize`?

File: code/game/objects/items/weapons/grenades/chem_grenade.dm [hunk 2]
```diff
@@ -41,8 +41,6 @@
 	if(stage > 1 && !active && clown_check(user))
 		user << SPAN_WARNING("You prime \the [name]!")
 
-		msg_admin_attack("[user.name] ([user.ckey]) primed \a [src]. (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)")
-
```
Comment (LethalGhost) [inline code/game/objects/items/weapons/grenades/chem_grenade.dm]: That's kind of unrelated change.

Comment (ITCups) [reply code/game/objects/items/weapons/grenades/chem_grenade.dm]: i love unrelated changes.
This message is not needed since missing igniter. It wont go off so there is no need for it.

File: code/game/objects/effects/effect_system.dm (hunk 1 above)
Comment (ITCups) [reply code/game/objects/effects/effect_system.dm]: is there a solid reason why it should be in Initialize ?

Comment (LethalGhost) [reply code/game/objects/effects/effect_system.dm]: All initialization code should go into `Initialize`/`LateInitialize` to prevent maploading lag and keep things controllable.

File: code/game/objects/items/weapons/grenades/chem_grenade.dm (hunk 2 above)
Comment (LethalGhost) [reply code/game/objects/items/weapons/grenades/chem_grenade.dm]: Happy for you, but all changes should be related to PR topic.
Or at least mentoinded in a description.

File: code/game/objects/effects/effect_system.dm (hunk 1 above)
Comment (ITCups) [reply code/game/objects/effects/effect_system.dm]: fixed

**Comment by ACCount12** [comment]
Does this PR fix the issue with burrows tearing off the piping, resulting in broken power/atmos/engine? They should either ignore the pipes, or only spawn in titles that don't have them.

**Comment by ITCups** [comment]
i think it does, previously it was damaging the tile it was on, i deleted that.

File: code/modules/power/sensors/powernet_sensor.dm [hunk 3]
```diff
@@ -167,7 +167,7 @@
 			APC_entry["total_load"] = reading_to_text(A.lastused_total)
 			// Hopefully removes those goddamn \improper s which are screwing up the UI
 			var/N = A.area.name
-			if(findtext(N, "�"))
+			if(findtext(N, "�"))
```
Comment (ACCount12) [inline code/modules/power/sensors/powernet_sensor.dm]: What is this change? Does the code still work?

File: code/game/turfs/simulated/floor_attackby.dm [hunk 4]
```diff
@@ -36,6 +36,13 @@
 			user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN*1.75) //This longer cooldown helps promote skill in melee combat by punishing misclicks a bit
 			return TRUE
 
+	for(var/atom/movable/A in src)
+		if(A.preventsTurfInteractions())
+			to_chat(user, SPAN_NOTICE("[A] is on the way."))
```
Comment (ACCount12) [inline code/game/turfs/simulated/floor_attackby.dm]: "in the way"

File: code/game/objects/structures/burrows.dm [hunk 5]
```diff
@@ -599,3 +634,20 @@ percentage is a value in the range 0..1 that determines what portion of this mob
 	//will be quieter and not travel as far
 	playsound(src, soundtype, maintenance ? volume*0.5 : volume, TRUE,maintenance ? -3 : 0)
 
+/obj/structure/burrow/examine()
+	..()
+	if(isSealed && recieving)
+		to_chat(usr, SPAN_WARNING("You can see movement behind cracks. Maybe they can be welded before whatever behind them break through."))
```
Comment (ACCount12) [inline code/game/objects/structures/burrows.dm]: Broken English again.
My attempt is "You can see something move behind the cracks. You should weld them shut before it breaks through.". Not perfect, but better already.

File: code/game/objects/effects/decals/weldable.dm [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+	name = "cracks"
+	desc = "Cracks on the tile. Perhaps a welding tool can help to get rid of them."
+	gender = PLURAL
+	icon = 'icons/obj/burrows.dmi'
+	icon_state = "cracks"
+	anchored = 1
+
+/obj/effect/decal/weldable/attackby(obj/item/I, mob/user)
+	if (I.has_quality(QUALITY_WELDING))
+		user.visible_message("[user] starts welding [src] with the [I]", "You start welding and [src] with the [I]")
```
Comment (ACCount12) [inline code/game/objects/effects/decals/weldable.dm]: "welding and"?
